src/store/useShapeDiverStoreParameters.ts

```diff
@@ -69,26 +71,59 @@ function createParameterExecutor<T>(sessionId: string, param: IGenericParameterD
 	};
 }
 
-function createGenericParameterExecutorForSession(session: ISessionApi) : IGenericParameterExecutor { 
+type DefaultExportsGetter = () => string[];
+type ExportResponseSetter = (response: IExportResponse) => void;
+
+function isFileParameter(param: IParameterApi<unknown>): param is IFileParameterApi {
+	return ("upload" in param);
+}
+
+function createGenericParameterExecutorForSession(session: ISessionApi, 
+	getDefaultExports: DefaultExportsGetter, exportResponseSetter: ExportResponseSetter) : IGenericParameterExecutor { 
 	
-	return (values) => {
+	return async (values) => {
 
 		// store previous values (we restore them in case of error)
 		const previousValues = Object.keys(values).reduce((acc, paramId) => {
 			acc[paramId] = session.parameters[paramId].value;
 			
 			return acc;
 		}, {} as { [paramId: string]: unknown});
+
+		// get ids of default exports that should be requested
+		const exports = getDefaultExports();
+
 		try {
 			// set values and call customize
 			Object.keys(values).forEach(id => session.parameters[id].value = values[id]);
-
-			return session.customize();
-		} 
+		
+			if (exports.length > 0) {
+				// prepare parameters (e.g. upload)
+				await Promise.all(Object.values(session.parameters)
+					.filter(isFileParameter)
+					.map(async p => {
+						const fileId = await p.upload();
+						p.value = fileId;
+					})
+				);
```

Comment:
@MajorMeerkatThe3rd a detailed question about file parameters: do we on purpose not set the parameter value as part of the `upload` function?

Reply:
Related task: https://shapediver.atlassian.net/browse/SS-7487